recipe-server/client/control_new/components/recipes/CloneRecipePage.js

```diff
@@ -0,0 +1,103 @@
+import { message } from 'antd';
+import autobind from 'autobind-decorator';
+import { Map } from 'immutable';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+import { Link, push as pushAction } from 'redux-little-router';
+
+import LoadingOverlay from 'control_new/components/common/LoadingOverlay';
+import RecipeForm from 'control_new/components/recipes/RecipeForm';
+import QueryRecipe from 'control_new/components/data/QueryRecipe';
+
+import { createRecipe as createAction } from 'control_new/state/recipes/actions';
+import { getRecipe } from 'control_new/state/recipes/selectors';
+import { getRouterParamAsInt } from 'control_new/state/router/selectors';
+
+@connect(
+  state => {
+    const recipeId = getRouterParamAsInt(state, 'recipeId');
+
+    return {
+      recipeId,
+      recipe: getRecipe(state, recipeId, new Map()),
+    };
+  },
+  {
+    push: pushAction,
+    createRecipe: createAction,
+  },
+)
+@autobind
+export default class CloneRecipePage extends React.Component {
+  static propTypes = {
+    createRecipe: PropTypes.func.isRequired,
+    recipeId: PropTypes.number.isRequired,
+    recipe: PropTypes.instanceOf(Map),
+  };
+
+  static defaultProps = {
+    recipe: null,
+  };
+
+  state = {
+    formErrors: undefined,
```

Comment:
I think this needs to be inside a constructor, like below, but I'm not actually sure. Is this ok?

```js
constructor(props) {
  super(props);
  this.state = { formErrors: undefined };
}
```